cmd/root/install/mssql-base.go

```diff
@@ -0,0 +1,324 @@
+		DefaultString: "mcr.microsoft.com",
+		Usage:         "Container registry",
+	})
+
+	addFlag(cmdparser.FlagOptions{
+		String:        &c.repo,
+		Name:          "repo",
+		DefaultString: repo,
+		Usage:         "Container repository",
+	})
+
+	addFlag(cmdparser.FlagOptions{
+		String:        &c.tag,
+		Name:          "tag",
+		DefaultString: "latest",
+		Usage:         "Tag to use, use get-tags to see list of tags",
+	})
+
+	addFlag(cmdparser.FlagOptions{
+		String:    &c.contextName,
+		Name:      "context-name",
+		Shorthand: "c",
+		Usage:     "Context name (a default context name will be created if not provided)",
+	})
+
+	addFlag(cmdparser.FlagOptions{
+		String:    &c.defaultDatabase,
+		Name:      "user-database",
+		Shorthand: "u",
+		Usage:     "Create a user database and set it as the default for login",
+	})
+
+	addFlag(cmdparser.FlagOptions{
+		Bool:  &c.acceptEula,
+		Name:  "accept-eula",
+		Usage: "Accept the SQL Server EULA",
+	})
+
+	addFlag(cmdparser.FlagOptions{
+		Int:        &c.passwordLength,
+		DefaultInt: 50,
+		Name:       "password-length",
+		Usage:      "Generated password length",
+	})
+
+	addFlag(cmdparser.FlagOptions{
+		Int:        &c.passwordMinSpecial,
+		DefaultInt: 10,
+		Name:       "password-min-special",
+		Usage:      "Minimum number of special characters",
+	})
+
+	addFlag(cmdparser.FlagOptions{
+		Int:        &c.passwordMinNumber,
+		DefaultInt: 10,
+		Name:       "password-min-number",
+		Usage:      "Minimum number of numeric characters",
+	})
+
+	addFlag(cmdparser.FlagOptions{
+		Int:        &c.passwordMinUpper,
+		DefaultInt: 10,
+		Name:       "password-min-upper",
+		Usage:      "Minimum number of upper characters",
+	})
+
+	addFlag(cmdparser.FlagOptions{
+		String:        &c.passwordSpecialCharSet,
+		DefaultString: "!@#$%&*",
+		Name:          "password-special-chars",
+		Usage:         "Special character set to include in password",
+	})
+
+	c.encryptPasswordFlag(addFlag)
+
+	addFlag(cmdparser.FlagOptions{
+		Bool:  &c.useCached,
+		Name:  "cached",
+		Usage: "Don't download image.  Use already downloaded image",
+	})
+
+	// BUG(stuartpa): SQL Server bug: "SQL Server is now ready for client connections", oh no it isn't!!
+	// Wait for "Server name is" instead!  Nope, that doesn't work on edge
+	// Wait for "The default language" instead
+	// BUG(stuartpa): This obviously doesn't work for non US LCIDs
+	addFlag(cmdparser.FlagOptions{
+		String:        &c.errorLogEntryToWaitFor,
+		DefaultString: "The default language",
+		Name:          "errorlog-wait-line",
+		Usage:         "Line in errorlog to wait for before connecting to disable 'sa' account",
+	})
+
+	addFlag(cmdparser.FlagOptions{
+		String:        &c.collation,
+		DefaultString: "SQL_Latin1_General_CP1_CI_AS",
+		Name:          "collation",
+		Usage:         "The SQL Server collation",
+	})
+}
+
+func (c *MssqlBase) Run() {
+	var imageName string
+
+	if !c.acceptEula && viper.GetString("ACCEPT_EULA") == "" {
```

Comment:
>if !c.acceptEula && viper.GetString("ACCEPT_EULA") == " [](http://example.com/codeflow?start=1&length=55)

viper has a way to bind its config values to flags. Do you see us using that technique such that code like this would only have to check in viper?

Reply:
Is I am using the viper feature to bind to env here ( err = viper.BindEnv("ACCEPT_EULA")), but no binding to the deep sub cmd flag.  The reason I didn't do that is the flag is deep down in the hierarchy, e.g. sqlcmd install mssql --flag, which would be a long variable name, someting like SQLCMD_INSTALL_MSSQL_ACCEPT_EULA I think)